### PR TITLE
Add support for calculations

### PR DESCRIPTION
**Pull request by chrischu**
Hi!

I thought it might be useful to have support for calculating values in Octopus.

One of the use cases we had in our own team was that we wanted to define an IP range per project but the actual IP via tenant. Currently we have a project variable "IP.Prefix" (e.g. `192.168.0`) and a tenant variable "IP.Suffix" (e.g. `5`) and they will be appended. However it would be easier for us to just have an "IP" project variable (e.g. `192.168.0.#{calc 10 + IP.Offset}`) and a "IP.Offset" variable in the tenants.

There are probably a multitude of other use cases for mathematical calculations.

For now the supported operators are only `+`, `-`, `/`, `*` but this could be extended quite easily.

### CODE REVIEW

**Comment by chrischu** [comment]
Mhh for some reason the build seems to fail, but I cannot access the TeamCity server to see what the actual error is. When I run the unit tests in my local environment I can see that there are some tests that are failing (but they also fail on the master branch...).

File: source/Octostache.Tests/CalculationFixture.cs
Comment (mjhilton) [inline source/Octostache.Tests/CalculationFixture.cs:14]: This is problematic, because of region formatting. 

[Half the world uses the decimal comma, and half use the decimal point](https://en.wikipedia.org/wiki/Decimal_separator#Usage_worldwide). Our build server happens to be configured in a region that uses a decimal point, so the build failure you saw was due to this test failing:

```
X Octostache.Tests.CalculationFixture.ConditionalIsSupported(expression: "3/2", expectedResult: "1,5")s]
Error Message:
 Expected string to be "1,5", but "1.5" differs near ".5" (index 1).
Stack Trace:
   at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message)
 at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
 at FluentAssertions.Execution.DefaultAssertionStrategy.HandleFailure(String message)
 at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
 at FluentAssertions.Primitives.StringEqualityValidator.ValidateAgainstMismatch()
 at FluentAssertions.Primitives.StringValidator.Validate()
 at FluentAssertions.Primitives.StringAssertions.Be(String expected, String because, Object[] becauseArgs)
 at Octostache.Tests.CalculationFixture.ConditionalIsSupported(String expression, String expectedResult) in 
```

**Review by mjhilton — COMMENTED** [review]
Details of why the build failed above.

File: source/Octostache.Tests/CalculationFixture.cs
Comment (chrischu) [reply source/Octostache.Tests/CalculationFixture.cs]: Thanks, I think I fixed it with my latest changes :).

**Comment by chrischu** [comment]
@mjhilton Hi! Is there any update on the review? The build is fixed for quite some time, so the only thing that blocks the merge is a review :).

**Review by mjhilton — APPROVED** [review]
This looks like a reasonable change with good tests around it. Thanks for your contribution, and sorry it took so long to provide a review.

**Comment by chrischu** [comment]
@mjhilton Thanks!

I tried to fix the last pending check (the license check), but I cannot seem to get the acceptance check working. I get to the page where I see the license (and disabled fields for my name, email, etc.) but there is no button or anything that would allow me to continue :O.

**Comment by mjhilton** [comment]
Thanks again for your contribution @chrischu, and for the crazy long wait time for me to action this. My apologies that it took so long. 

We've internally built Octostache v3.3.0, and I've raised the PRs to integrate the new version [into Calamari](https://github.com/OctopusDeploy/Calamari/pull/984), [Octopus Server](https://github.com/OctopusDeploy/OctopusDeploy/pull/16369) (🔒 private repo), and to [document the changes](https://github.com/OctopusDeploy/docs/pull/1781).

We'll be targeting the 2023 Q1 release of Octopus Server for this change. I'll let you know via a comment on this PR when it's released, and the exact Octopus Server version number it'll be available from.

**Comment by mjhilton** [comment]
I [raised an enhancement issue](https://github.com/OctopusDeploy/Issues/issues/8096) to cover including this change in Octopus Server. If you follow that issue, it will be automatically updated with details of which version the change appears in.

**Comment by chrischu** [comment]
Thanks for the update! :)

**Pull request merged**
